composer/trainer/trainer.py

```diff
@@ -1161,10 +1179,56 @@ def __init__(
         assert not isinstance(device_train_microbatch_size, str)
 
         # Distributed
-        if deepspeed_config is not None or fsdp_config is not None or dist.get_world_size() > 1:
+        if fsdp_config is not None:
+            warnings.warn(
+                VersionedDeprecationWarning(
+                    "fsdp_config is deprecated. Please use parallelism_config['fsdp'] instead.",
+                    remove_version='0.26.0',
+                ),
+            )
+            if parallelism_config is None:
+                parallelism_config = {}
+            if parallelism_config.get('fsdp') is not None:
+                raise ValueError(
+                    'fsdp_config is specified in both fsdp_config and parallelism_config. Please specify it in only in parallelism_config.',
+                )
+            parallelism_config['fsdp'] = fsdp_config
+        if not fsdp_auto_wrap:
+            warnings.warn(
+                VersionedDeprecationWarning(
+                    "fsdp_auto_wrap=False is deprecated. Please use parallelism_config['fsdp']['auto_wrap'] instead.",
+                    remove_version='0.26.0',
+                ),
+            )
+            if parallelism_config is None:
+                parallelism_config = {}
+            if parallelism_config.get('fsdp') is None:
+                parallelism_config['fsdp'] = {}
+            parallelism_config['fsdp']['auto_wrap'] = fsdp_auto_wrap
+        if parallelism_config is not None:
+            # Set defaults and create shallow copies of configs to avoid changing user's config
+            parallelism_config = {**parallelism_config}
+            if parallelism_config.get('fsdp', None) is not None:
+                parallelism_config['fsdp'] = set_fsdp_default({**parallelism_config['fsdp']})
+            if parallelism_config.get('tp', None) is not None:
+                parallelism_config['tp'] = {**parallelism_config['tp']}
+            # Remove empty configs
+            for key in list(parallelism_config.keys()):
+                if parallelism_config[key] == None:
+                    del parallelism_config[key]
+            if len(parallelism_config) == 0:
+                parallelism_config = None
+        if deepspeed_config is not None and parallelism_config is not None:
+            raise ValueError(
+                'Both deepspeed_config and parallelism_config are specified but incompatible. Please specify only one.',
+            )
+        if deepspeed_config is not None or parallelism_config is not None or dist.get_world_size() > 1:
             # Deepspeed and FSDP both require torch.distributed to be initialized, even if the world size is 1
             # And torch.distributed is always required for multi-rank training
             dist.initialize_dist(device, dist_timeout)
+        if parallelism_config is not None or deepspeed_config is not None:
```

Comment:
Do you want to patch pytorch for deep speed usage? I guess might as well?

Reply:
Unintentional, fixed